godot-core/src/builtin/aabb.rs

```diff
@@ -43,32 +43,346 @@ impl Aabb {
     }
 
-    /* Add in when `Aabb::abs()` is implemented.
+    /// Returns the normalized longest axis of the AABB.
+    ///
+    /// _Godot equivalent: `AABB.get_longest_axis()`_
+    pub fn longest_axis(&self) -> Vector3 {
+        match self.longest_axis_index() {
+            Vector3Axis::X => Vector3::RIGHT,
+            Vector3Axis::Y => Vector3::UP,
+            Vector3Axis::Z => Vector3::BACK,
+        }
+    }
+
+    /// Returns the index of the longest axis of the AABB (according to Vector3's AXIS_* constants).
+    ///
+    /// _Godot equivalent: `AABB.get_longest_axis_index()`_
+    pub fn longest_axis_index(&self) -> Vector3Axis {
+        self.size.max_axis_index()
+    }
+
+    /// Returns the scalar length of the longest axis of the AABB.
+    ///
+    /// _Godot equivalent: `AABB.get_longest_axis_size()`_
+    pub fn longest_axis_size(&self) -> real {
+        self.size.x.max(self.size.y.max(self.size.z))
+    }
+
+    /// Returns the normalized shortest axis of the AABB.
+    ///
+    /// _Godot equivalent: `AABB.get_shortest_axis()`_
+    pub fn shortest_axis(&self) -> Vector3 {
+        match self.shortest_axis_index() {
+            Vector3Axis::X => Vector3::RIGHT,
+            Vector3Axis::Y => Vector3::UP,
+            Vector3Axis::Z => Vector3::BACK,
+        }
+    }
+
+    /// Returns the index of the shortest axis of the AABB (according to Vector3::AXIS* enum).
+    ///
+    /// _Godot equivalent: `AABB.get_shortest_axis_index()`_
+    pub fn shortest_axis_index(&self) -> Vector3Axis {
+        self.size.min_axis_index()
+    }
+
+    /// Returns the scalar length of the shortest axis of the AABB.
+    ///
+    /// _Godot equivalent: `AABB.get_shortest_axis_size`_
+    pub fn shortest_axis_size(&self) -> real {
+        self.size.x.min(self.size.y.min(self.size.z))
+    }
+
+    /// Returns the support point in a given direction. This is useful for collision detection algorithms.
+    ///
+    /// _Godot equivalent: `AABB.get_support()`_
+    pub fn support(&self, dir: Vector3) -> Vector3 {
+        let half_extents = self.size * 0.5;
+        let ofs = self.position + half_extents;
+
+        let x = if dir.x > 0.0 {
+            half_extents.x
+        } else {
+            -half_extents.x
+        };
+
+        let y = if dir.y > 0.0 {
+            half_extents.y
+        } else {
+            -half_extents.y
+        };
+
+        let z = if dir.z > 0.0 {
+            half_extents.z
+        } else {
+            -half_extents.z
+        };
+
+        Vector3::new(x, y, z) + ofs
+    }
+
+    /// Returns `true` if the AABB overlaps with `b` (i.e. they have at least one point in common).
+    ///
+    /// If `include_borders` is `true`, they will also be considered overlapping if their borders touch, even without intersection.
+    ///
+    /// _Godot equivalent: `AABB.intersects(AABB b, bool include_borders)`_
+    #[inline]
+    pub fn intersects(&self, &b: &Aabb, include_borders: bool) -> bool {
```

Comment:
Maybe this should be split into two functions:
* `intersects`
* `intersects_excl_borders`

The `true`/`false` literal on call site is not very expressive otherwise.

I chose the overload _including_ borders as the default (with short name), because both `Aabb::intersection()` and `Rect2i::intersection()` include borders.

Reply:
However, in Godot, borders are excluded by default, is that a problem?

Reply:
Well, now I'm confused... I am sure that there was this `include_border`, but it seems like I have made a mistake. I'll work on that.
However, whether or not the border is included is very vague in the [documentation](https://docs.godotengine.org/en/latest/classes/class_aabb.html#class-aabb-method-intersects):

![image](https://github.com/godot-rust/gdext/assets/75273490/b7e503aa-d181-4a89-ac21-bb92eb4f477b)

Reply:
`Aabb` and `Rect2` are semantically very closely related. We should provide the same API where possible, even if Godot itself doesn't.

One reason why we implement this in Rust is that we can provide a more idiomatic interface to Rust developers, meaning:
* consistency
* expressivity
* better error handling
* use of Rust language features